R/helper.R

```diff
@@ -1,23 +1,69 @@
-demo_raster = function(x, cells, dimension) {
-  data = matrix(c(stats::rnorm(cells / 2, 0, 1), stats::rnorm(cells / 2, 1, 1)),
-    nrow = dimension
-  )
-  raster::raster(data, xmn = 0, xmx = dimension, ymn = 0, ymx = dimension)
+#' @title Generate Demo Raster
+#'
+#' @description
+#' Generates a square demo [terra::SpatRaster].
+#'
+#' @param dimension (`integer(1)`)
+#' xy dimension of raster
+#'
+#' @export
+demo_raster = function(dimension) {
+  assert_int(dimension, lower = 2)
+  data = matrix(c(stats::rnorm(floor(dimension^2 / 2), 0, 1),
+    stats::rnorm(ceiling(dimension^2 / 2), 1, 1)), nrow = dimension)
+  terra::rast(data)
 }
 
-demo_stack = function(size = 1000000000, layers = 5) {
-  dimension = floor(sqrt(size / 8 / layers))
-  if (dimension %% 2 == 1) {
-    dimension = dimension - 1
-  }
-  cells = dimension^2
+#' @title Generate Demo Raster Stack
+#'
+#' @description
+#' Generates a square demo [terra::SpatRaster] stack.
+#'
+#' @param size (`integer(1)`)\cr
+#' Size of raster stack in megabyte.
+#' @param layer (`integer(1)`)\cr
+#' Number of layers.
+#'
+#' @export
+demo_stack = function(size = 500, layers = 5) {
+  assert_int(size, lower = 1)
+  assert_int(layers, lower = 1)
 
-  response = matrix(rep(c(0, 1), each = cells / 2),
-    nrow = dimension
-  )
-  response = list(raster::raster(response, xmn = 0, xmx = dimension, ymn = 0, ymx = dimension))
+  dimension = floor(sqrt(size / layers * 1e+06 / 8))
+  raster_features = replicate(layers - 1, demo_raster(dimension))
+  raster_response = rast(matrix(c(rep(0, floor(dimension^2 / 2)),
+    rep(1, ceiling(dimension^2 / 2))), nrow = dimension))
+  raster = rast(c(raster_features, list(raster_response)))
+  names(raster) = c(paste0("x_", 1:(layers - 1)), "y")
+  raster
+}
+
+
+#' @title Split Raster Into Chunks
+#'
+#' @description
+#' Splits raster into chunks.
+#'
+#' @param raster ([terra::SpatRaster])\cr
+#' Raster to be split into chunks.
+#' @param chunksize
+#' Raster chunk size in megabyte.
+#'
+#' @export
+block_size = function(raster, chunksize) {
```

Comment:
I am wondering if we could come up with some smart default based on the available hardware resources at hand (mem + cores).
We could print the selected resources during function invoke and suggest the user to adapt these to their needs.